### DISCUSSION TITLE
PulseTime is counting down even with with channel turned off

### DISCUSSION THREAD

**Discussion by soeffi** [arendst/Tasmota#15385] [Q&A] [answered]
All,

after setting PulseTime to a new value, PulseTime begins counting down (as displayed by PulseTime.Remaining), even when channel is powered off.

Expected behavior: With channel powered on, this is the expected behavior. With channel powered off, PulseTime.Remaining should report 0 until powered on, imho.

Note: After a manual power on + power off cycle, PulseTime.Remaining reports 0 as expected.

I'd consider this a bug (low priority, but still annoying / confusing, as I display remaining time along with power status, which makes no sense to have a countdown with the device powered off) - or anyone with an explanation for this?

**Comment by barbudor** [comment, marked as answer]
This doesn't seems to be a bug but feature.
Looking at the code, the PulseTime command not only store the value in the settings but deliberately starts a PulseTime count down.

**Reply by soeffi** [reply]
Which makes sense when power state is on. Imho, however, there should be a check for the power state to not start the count down in "off" state.
Does that make sense?

**Reply by soeffi** [reply]
Apologize for bringing this up again, but I do think the answer falls a little short.

> This doesn't seems to be a bug but feature. Looking at the code, the PulseTime command not only store the value in the settings but deliberately starts a PulseTime count down.

I tried to understand the code as well, I guess you refer to the call to SetPulseTimer https://github.com/arendst/Tasmota/blob/c988ba16451186c55873793978e7730619a25074/tasmota/tasmota_support/support_command.ino#L979 which starts the countdown?

I'm pretty sure, that this was added here to reset the count down to the new PulseTime value when power state is on. I do think, however, there should be a a check for power state being "on" in front of this. For someone familiar with the code it's probably straight-forward to figure out the correct clause like
`if ( "power state is on" ) SetPulseTimer(...)`

Anyone willing to help me out and getting this fixed?

**Reply by sfromis** [reply]
Why do you "need" a fix? Sure, the remaining time is counting down after initially doing the (in principle one-time) config, if power was already on, but I saw no harm done. No such effect seen if power was off before starting, at least with current Tasmota version.

**Reply by soeffi** [reply]
Well, I must admit that it comes down to "aesthetics" and a bit of confirmation bias on my end when reading the docs, as I was expecting something else here.

My understanding of "remaining pulsetime" obtained by the PulseTime command is "how long will that device still be turned on before it turns off", with an implicit expectation that pulsetime > 0 means that the device is actually on in fact.

As I said, it's not a functional issues, but counter-intuitive in my visualization to see PulseTime counting down with the device actually being off (I'd expect something to happen when the count down reaches zero, but that's obviously not the case).

Anyway, I will "fix" this for my use-case by displaying remaining pulsetime only when the device is on, which is ok for me. Changing tasmota code was just a suggestion to make things more intuitive, but maybe that's just me.

**Reply by soeffi** [reply]
> after initially doing the (in principle one-time) config

I change PulseTime regularly (irrigation time based on weather reports), so not only one-time config in my case.

**Reply by sfromis** [reply]
Again, with a device on current Tasmota (not an older version), setting a PulseTime while the output power was off, I did not see remaining time counting down.

**Reply by soeffi** [reply]
On 12.0.2:

Channel1 is off:
```
23:01:31.964 CMD: Power1
23:01:31.970 MQT: stat/regenmacher/RESULT = {"POWER1":"OFF"}
23:01:31.973 MQT: stat/regenmacher/POWER1 = OFF
```

Set PulseTime:
```
23:01:36.426 CMD: PulseTime1 200
23:01:36.432 MQT: stat/regenmacher/RESULT = {"PulseTime1":{"Set":200,"Remaining":200}}
```

check remaining time::
```
23:01:39.529 CMD: PulseTime1
23:01:39.535 MQT: stat/regenmacher/RESULT = {"PulseTime1":{"Set":200,"Remaining":196}}
23:01:41.129 CMD: PulseTime1
23:01:41.135 MQT: stat/regenmacher/RESULT = {"PulseTime1":{"Set":200,"Remaining":195}}
23:01:43.583 CMD: PulseTime1
23:01:43.588 MQT: stat/regenmacher/RESULT = {"PulseTime1":{"Set":200,"Remaining":192}}
```
